test/LogTest.php

```diff
@@ -0,0 +1,222 @@
+<?php
+/**
+ * This file is part of OPUS. The software OPUS has been originally developed
+ * at the University of Stuttgart with funding from the German Research Net,
+ * the Federal Department of Higher Education and Research and the Ministry
+ * of Science, Research and the Arts of the State of Baden-Wuerttemberg.
+ *
+ * OPUS 4 is a complete rewrite of the original OPUS software and was developed
+ * by the Stuttgart University Library, the Library Service Center
+ * Baden-Wuerttemberg, the Cooperative Library Network Berlin-Brandenburg,
+ * the Saarland University and State Library, the Saxon State Library -
+ * Dresden State and University Library, the Bielefeld University Library and
+ * the University Library of Hamburg University of Technology with funding from
+ * the German Research Foundation and the European Regional Development Fund.
+ *
+ * LICENCE
+ * OPUS is free software; you can redistribute it and/or modify it under the
+ * terms of the GNU General Public License as published by the Free Software
+ * Foundation; either version 2 of the Licence, or any later version.
+ * OPUS is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details. You should have received a copy of the GNU General Public License
+ * along with OPUS; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ * @category    Test
+ * @package     OpusTest
+ * @author      Kaustabh Barman <barman@zib.de>
+ * @copyright   Copyright (c) 2020, OPUS 4 development team
+ * @license     http://www.gnu.org/licenses/gpl.html General Public License
+ */
+
+namespace OpusTest;
+
+use Opus\Log;
+
+class LogTest extends \PHPUnit_Framework_TestCase
+{
+
+    private $logFile;
+
+    public function tearDown()
+    {
+        if (is_resource($this->logFile)) {
+            fclose($this->logFile);
+        }
+
+        parent::tearDown();
+    }
+
+    public function testSetLevel()
+    {
+        $opusLog = $this->getOpusLog();
+        $opusLog->setLevel(\Zend_Log::DEBUG);
+        $debugMessage = 'Debug level message from testSetLevel';
+        $opusLog->debug($debugMessage);
+        $content = $this->readLog();
+
+        $level = $opusLog->getLevel();
+
+        $this->assertContains($debugMessage, $content);
+        $this->assertEquals(\Zend_Log::DEBUG, $level);
+    }
+
+    public function testSetLevelForFilterDisabling()
+    {
+        $opusLog = $this->getOpusLog();
+        $opusLog->setLevel(null);
+
+        $debugMessage = 'Debug level message from testSetLevelWithNullLevel';
+        $opusLog->debug($debugMessage);
+
+        $content = $this->readLog();
+
+        $this->assertContains($debugMessage, $content);
+    }
+
+    public function testSetLevelToNullHavingCustomLevels()
```

Comment:
Is unclear what we are actually testing here.

Reply:
I mean does `setLevel(null)` really matter here? I know it originates from an earlier implementation where it did matter.

Reply:
That helps, but it is the wrong style of comment here. Should be `/** ... */`. Even better would be changing the name of the test accordingly and get rid of the comment. We are not really testing `setLevel`. We have other tests for that. We are testing the behavior of custom messages in a disabled filter.